aptos-node/src/tests.rs

```diff
@@ -51,3 +51,39 @@ fn test_mutual_authentication_validators() {
 fn test_aptos_vm_does_not_have_test_natives() {
     aptos_vm::natives::assert_no_test_natives(crate::utils::ERROR_MSG_BAD_FEATURE_FLAGS)
 }
+
+#[test]
+fn test_config_override_merge() {
```

Comment:
noice!

Although it would be very helpful to add a test for `create_single_node_test_config` since that actually has more logic than the merge itself

Ideally the merge logic is covered by their own unittests so this is really just testing their library which is less useful of a test